File: Swift-C++-IR-Comparison/test/closures/Swift/AutoclosureFunctioncallExplicit.swift
// As an optimization, Swift may instead capture and store a copy of
//  a value if that value isn’t mutated by a closure, and if the value
//  isn’t mutated after the closure is created.

// Swift also handles all memory management involved in disposing of
// variables when they’re no longer needed.

@main
struct MyMain {

    static func main() {
        var customersInLine = [1, 2, 3, 4, 5, 6, 42]

        func serve(customer customerProvider: @autoclosure () -> Int) {
            let tmp = customerProvider()
        }
        serve(customer: customersInLine.remove(at: 0))
    }
}

File: AutoTranspiledTests/try-0/AutoclosureFunctioncallExplicit.cpp
 
 #include <iostream> 
using namespace std; 
int main()  
{    
    int customersInLine[7] = {1, 2, 3, 4, 5, 6, 42}; 
 
    // serve function 
    auto serve = [&]() -> int             
    { 
        int tmp = customersInLine[0]; 
        for (int i = 0; i <= 6; i++) 
            customersInLine[i] = customersInLine[i + 1]; 
        return tmp; 
    }; 
     
    auto customerProvider = serve(); 
    
    cout << customerProvider; 
 
    return 0; 
}